tools/codegen/pkg/modelgen/model.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package modelgen
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+
+	tmpl "istio.io/mixer/tools/codegen/pkg/template_extension"
+)
+
+const FullNameOfExprMessage = "istio_mixer_v1_config_template.Expr"
+const FullNameOfExprMessageWithPtr = "*" + FullNameOfExprMessage
+
+type (
+	// Data model to be used to generate code for istio/mixer
+	Model struct {
+		// top level fields
+		Name        string
+		PackageName string
+		VarietyName string
+
+		// imports
+		Imports []string
+
+		// Constructor fields
+		ConstructorFields []fieldInfo
+
+		// Warnings/Errors in the Template proto file.
+		Diags []Diag
+	}
+
+	fieldInfo struct {
+		Name string
+		Type typeInfo
+	}
+
+	typeInfo struct {
+		Name string
+	}
+)
+
+// Create creates a Model object.
+func Create(parser *FileDescriptorSetParser) (*Model, error) {
+	var templateProto *FileDescriptor
+
+	if template, diags := getTmplFileDesc(parser.allFiles); len(diags) != 0 {
+		return nil, createError(diags)
+	} else {
```

Comment:
you could bump the `template, diags := getTmplFileDesc(parser.allFiles)` up a line and avoid the else (and also avoid the need for `templateProto` altogether).